src/extend-expect.ts

```diff
@@ -45,7 +45,11 @@ expect.extend(
         ...matcherArgs: unknown[]
       ) {
         const serverPromise = createClientRuntimeServer();
-        if (typeof elementHandle !== 'object' || !elementHandle?.asElement()) {
+        if (
+          typeof elementHandle !== 'object' ||
+          // eslint-disable-next-line @cloudfour/typescript-eslint/no-unnecessary-condition
+          !elementHandle?.asElement?.()
+        ) {
```

Comment:
Nit: I found this kind of difficult to parse this conditional when it was split across multiple lines with a comment in the middle. I wonder if something like this would be easier to read?

```suggestion
        // eslint-disable-next-line @cloudfour/typescript-eslint/no-unnecessary-condition
        const hasAsElementMethod = elementHandle?.asElement?.();
        if (typeof elementHandle !== 'object' || !hasAsElementMethod) {
```

I don't feel super strongly about this though, so if that doesn't work as expected or has other downsides feel free to ignore.

Reply:
I extracted an `isElementHandle` function

Reply:
Oh perfect, thanks